R/connect.R

```diff
@@ -143,8 +144,8 @@ CLIENT_RESERVED     <- 16384
 # Old flag for 4.1 protocol
 
 #' @export
-CLIENT_SECURE_CONNECTION <- 32768
-# New 4.1 authentication
+CLIENT_RESERVED2    <- 32768
```

Comment:
Would this flag still be useful for connecting with older servers? We could keep it as is and add a note to the documentation instead.

I'm fine with removing if we're sure that this flag is now truly useless in all conceivable settings.

Reply:
I have no problem leaving it in - this flag was from protocol 4.1, so has been deprecated for a pretty long time.  But, CLIENT_RESERVED2 is not really being used for anything either.   So, I would be fine with leaving it, but definitely with a big note.  FWIW, none of the mysql libraries after 5.6 (including any mariadb) have it defined (it was deprecated and removed in 5.7).  And it wouldn't help with connecting to an older server as it's the local client library that would need to deal with that flag.  So, unless someone is still using the mysql 5.6 client library, it's not going to work.
We could also do something simple like:
```
CLIENT_RESERVED2 <- 32678
## For backward compatibility with mysql client libraries 5.6 and earlier and using protocol41
CLIENT_SECURE_CONNECTION <- CLIENT_RESERVED2
```
I guess I was just under the impression you were trying to keep this interface more current since you removed using `mysql_set_ssl()` as it was deprecated in mysql 8.0.

Let me know whether you want me to add that definition.

Reply:
5.6 is at end of life, I guess it's fine to remove it.